src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs

```diff
@@ -77,7 +92,13 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
                 throw new ArgumentNullException(nameof(request));
             }
 
-            return _policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
+            var policy = ResolvePolicy(request);
+            return policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
+        }
+
+        private IAsyncPolicy<HttpResponseMessage> ResolvePolicy(HttpRequestMessage request)
```

Comment:
@rynowak this is great, I think it solves most of the things I was concerned about.